addons/payment_adyen/tests/test_adyen.py

```diff
@@ -68,3 +69,17 @@ def test_send_refund_request(self):
             msg="The acquirer reference of the refund transaction should different from that of "
                 "the source transaction."
         )
+
+    @mute_logger('odoo.addons.payment_adyen.models.payment_transaction')
+    def test_incomplete_address(self):
+
+        test_partner = self.env['res.partner'].create({
+            'name': 'Dummy Partner',
+            'email': 'norbert.buyer@example.com',
+            'phone': '0032 12 34 56 78',
+        })
+
+        test_address = payment_utils_adyen.format_partner_address(test_partner)
+        for key in test_address:
+            if key in ('city', 'country', 'stateOrProvince', 'street'):
+                self.assertTrue(test_address[key])
```

Comment:
You are testing that if a key is set, and it is one of ('city', 'country', 'stateOrProvince', 'street'), we check that is has a non-falsy value. I'd rather check that, for all keys in ('city', 'country', 'stateOrProvince', 'street'), it must be set and have a non-falsy value.